### PR TITLE
Adding data coverage snapshot script

### PR DESCRIPTION
**Pull request by ghop02**
This adds in the script to generate the details about coverage that the data coverage table will be using.  

Runs a script that pulls in the latest summary file for each region type and aggregates based on json keys.

### CODE REVIEW

File: src/screens/DataApi/coverage-snapshot.json [hunk 1]
```diff
@@ -0,0 +1,632 @@
+[
+  {
+    "name": "riskLevels.overall",
```
Comment (mikelehen) [inline src/screens/DataApi/coverage-snapshot.json:3]: I wonder if this should be a map of metric=>metric coverage rather than an array, since the order presumably isn't significant?  But if this is actually easier to consume, that's fine.  Mostly just curious.

**Review by mikelehen — CHANGES_REQUESTED** [review]
In general I think this looks fine, but I found the code a bit hard to follow and I think it could use more comments at minimal.  Can you run through the comments and tackle whatever resonates / feels easy enough? Thanks!

File: scripts/update_coverage_snapshot.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
+import yargs from 'yargs';
+import fs from 'fs-extra';
+import { AggregationLevel } from '../src/api/schema/RegionSummary';
+import { RegionType } from '../src/common/regions';
+import { fetchSummariesForRegionType } from '../src/common/utils/model';
+import _ from 'lodash';
+
+interface MetricRecord {
```
[the remark this answers is not in the document]
Comment (ghop02) [reply scripts/update_coverage_snapshot.ts:8]: Yeah I struggled with the naming here

File: scripts/update_coverage_snapshot.ts [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+interface CoverageDetails {
+  totalRegions: number;
+  regionsAvailable: number;
+  totalStates: number;
+  statesAvailable: number;
+}
+
+export interface MetricCoverage {
```
[the remark this answers is not in the document]
Comment (ghop02) [reply scripts/update_coverage_snapshot.ts:24]: ah nope - part of the reason these are here is because I haven't added in the UI part where I was initially taking these from

File: scripts/update_coverage_snapshot.ts [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    `${outputFolder}/coverage-snapshot.json`,
+    coverageDetails,
+    {
+      spaces: 2,
+    },
+  );
+}
+
+if (require.main === module) {
+  main('src/screens/DataApi')
```
[the remark this answers is not in the document]
Comment (ghop02) [reply scripts/update_coverage_snapshot.ts:138]: ah meant to clean this up

**Review by mikelehen — APPROVED** [review]
